lib/livebook/file_system.ex

```diff
@@ -20,6 +20,13 @@ defprotocol Livebook.FileSystem do
   """
   @type path :: String.t()
 
+  @typedoc """
+  A unique identifier generated by the livebook for config storage purposes.
+
+  In case of the local filesystem, the id resolves to a `:local` atom.
+  """
+  @type id :: String.t() | :local
```

Comment:
Since the file system itself doesn't carry any id, I'd move it to `Livebook.Settings`, where we introduce it, wdyt?

Reply:
Sure, makes sense